### PR TITLE
Proposer RPC: GetBlock for Deneb

### PR DESCRIPTION
**Pull request by terencechain**
This PR updates Prysm RPC `Getblock` implementation for Deneb:

   - Use `build-block-parallel` as default. It should be defaulted by Deneb even before that. This makes the refactor cleaner and avoids time wasted on code that will be deleted later
   - `getEmptyBlock` returns empty deneb block
   - `getLocalPayload` -> `getLocalPayloadAndBlobs`. Returns blobs bundle alongside execution data
   - `setKzgCommitments` to set the KZG commitments to the block body
   - Helper `blobsBundleToSidecar` to convert `blobsBundle` to `blobSidecars`

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deneb_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	bundle := &enginev1.BlobsBundle{KzgCommitments: kcs, Proofs: proofs, Blobs: blobs}
+
+	sidecars, err := blobsBundleToSidecar(bundle, b.Block())
+	require.NoError(t, err)
+
+	require.Equal(t, len(sidecars), 3)
+
+	r, err := b.Block().HashTreeRoot()
+	require.NoError(t, err)
+	for i := 0; i < len(sidecars); i++ {
```
[the remark this answers is not in the document]
Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_deneb_test.go:54]: There is a length check before HTR: `require.Equal(t, len(sidecars), 3)`. I'll move it closer

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_test.go [hunk 2]
```diff
@@ -430,6 +430,126 @@ func TestServer_GetBeaconBlock_Capella(t *testing.T) {
...
+		PrevRandao:    random,
+		BaseFeePerGas: make([]byte, fieldparams.RootLength),
+		BlockHash:     make([]byte, fieldparams.RootLength),
+		Transactions:  make([][]byte, 0),
+		ExtraData:     make([]byte, 0),
+		BlockNumber:   1,
+		GasLimit:      2,
+		GasUsed:       3,
+		Timestamp:     uint64(timeStamp.Unix()),
+		ExcessDataGas: make([]byte, fieldparams.RootLength),
```
Comment (james-prysm) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_test.go:513]: do you want to update the payload first and this afterwards or merge this first.

Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_test.go:513]: good point, probably easier to wait for payload first, but doesn't really matter : )

**Review by james-prysm — APPROVED** [review]

**Comment by terencechain** [comment]
Will update the Payload fields all in one go

**Pull request merged**